core/icons/mutator_icon.ts

```diff
@@ -0,0 +1,358 @@
+/**
+ * @license
+ * Copyright 2023 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as goog from '../../closure/goog/goog.js';
+goog.declareModuleId('Blockly.Mutator');
+
+import {Abstract} from '../events/events_abstract.js';
+import type {Block} from '../block.js';
+import {BlockChange} from '../events/events_block_change.js';
+import {BlocklyOptions} from '../blockly_options.js';
+import {BlockSvg} from '../block_svg.js';
+import type {Connection} from '../connection.js';
+import {Coordinate} from '../utils/coordinate.js';
+import * as dom from '../utils/dom.js';
+import * as eventUtils from '../events/utils.js';
+import {IHasBubble} from '../interfaces/i_has_bubble.js';
+import {Icon} from './icon.js';
+import {MiniWorkspaceBubble} from '../bubbles/mini_workspace_bubble.js';
+import {MUTATOR_TYPE} from './icon_types.js';
+import {Rect} from '../utils/rect.js';
+import {Size} from '../utils/size.js';
+import {Svg} from '../utils/svg.js';
+import type {WorkspaceSvg} from '../workspace_svg.js';
+
+export class MutatorIcon extends Icon implements IHasBubble {
+  /** The type string used to identify this icon. */
+  static readonly TYPE = MUTATOR_TYPE;
```

Comment:
See previous comment about dealing with types.

Reply:
In an ideal world I want to get https://github.com/google/blockly/issues/7126 in by the end of the quarter, so that it wouldn't be a breaking change. 

In the case that this is not possible, all of the `IIcon` implementations will have a `static readonly TYPE: string` property, which can be passed to `getIcon`, `removeIcon`, etc. I think this is a reasonable implementation, even if it's not ideal from a type perspective.

I don't think the advantage of having it be a `static readonly TYPE: IconType` in the case of not implementing #7126 by EOQ is worth taking the time to implement it.

So my plan is to leave as-is for now (since this gets everything to build) then complete #7126 after #7115, #7128, and #7112 are merged.

IDK if that answers your comment :P but hopefully it clarifies!